sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubRemoteCredentialSupportTest.java

```diff
@@ -0,0 +1,181 @@
+        SecHubRemoteCredentialConfiguration configuration = supportToTest.getRemoteCredentialConfigurationFromJSONString(json);
+        String location = "example";
+        String type = "git";
+
+        /* execute */
+        List<SecHubRemoteCredentialContainer> credentialsLocation = supportToTest.resolveCredentialsForLocation(configuration, location);
+        List<SecHubRemoteCredentialContainer> credentialsType = supportToTest.resolveCredentialsForLocation(configuration, location, type);
+
+        /* test */
+        assertEquals(1, credentialsLocation.size());
+        assertEquals(0, credentialsType.size());
+
+        SecHubRemoteCredentialContainer container1 = credentialsLocation.get(0);
+        assertNotNull(container1);
+        assertEquals("userX", container1.getUser());
+        assertEquals("tokenX", container1.getPassword());
+
+    }
+
+    @Test
+    void match_remote_credentials_pattern_by_location() {
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_remote_credentials_complex_config.json"));
+        SecHubRemoteCredentialConfiguration configuration = supportToTest.getRemoteCredentialConfigurationFromJSONString(json);
+        String location1 = "https://github.com/username/project";
+        String location2 = "https://example.url.com/examples";
+        String location3 = "unknown.pattern";
+
+        /* execute */
+        List<SecHubRemoteCredentialContainer> credentials1 = supportToTest.resolveCredentialsForLocation(configuration, location1);
+        List<SecHubRemoteCredentialContainer> credentials2 = supportToTest.resolveCredentialsForLocation(configuration, location2);
+        List<SecHubRemoteCredentialContainer> credentials3 = supportToTest.resolveCredentialsForLocation(configuration, location3);
+
+        /* test */
+        assertEquals(1, credentials1.size());
+        assertEquals(3, credentials2.size());
+
+        SecHubRemoteCredentialContainer container1 = credentials1.get(0);
+        assertNotNull(container1);
+        assertEquals("example-user", container1.getUser());
+
+        SecHubRemoteCredentialContainer container2 = credentials2.get(0);
+        assertNotNull(container2);
+        assertEquals("user2", container2.getUser());
+
+        SecHubRemoteCredentialContainer container2b = credentials2.get(1);
+        assertNotNull(container2b);
+        assertEquals("user3", container2b.getUser());
+
+        SecHubRemoteCredentialContainer container2c = credentials2.get(2);
+        assertNotNull(container2c);
+        assertEquals("userX", container2c.getUser());
+
+        assertTrue(credentials3.isEmpty());
+    }
+
+    @Test
+    void match_remote_credentials_pattern_by_type() {
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_remote_credentials_complex_config.json"));
```

Comment:
Normally i am a fan of loading test data, but sometimes it can become difficult to understand whats happening inside test.

Sometimes (when data is not too much) it is just better to use the java 17 feature "text blocks"

and do something like this:

```java
String json = """
{
  "credentials": [
    {
      "user" : "example-user",
      "password" : "example-api-token-or-password",
      "remotePattern" : ".*github.com*",
      "type" : "example-type"
    },
    {
      "user" : "user2",
      "password" : "password2",
      "remotePattern" : ".*example*",
      "type" : "git"
    },
    {
      "user" : "user3",
      "password" : "token3",
      "remotePattern" : ".*example.url.com*",
      "type" : "docker,git"
    },
    {
      "user" : "userX",
      "password" : "tokenX",
      "remotePattern" : ".*example.*",
      "type" : "binary"
    }
  ]
}
"""
```

Doing this, some tests could maybe easier to understand